app/assets/javascripts/discourse/app/controllers/sidebar-section-form.js

```diff
@@ -29,12 +29,32 @@ class Section {
   }
 
   get validTitle() {
-    return !isEmpty(this.title) && this.title.length <= 30;
+    return !this.#blankTitle && !this.#tooLongTitle;
+  }
+
+  get invalidTitleMessage() {
+    if (this.title === undefined) {
+      return;
+    }
+    if (this.#blankTitle) {
+      return I18n.t("sidebar.sections.custom.title.validation.blank");
+    }
+    if (this.#tooLongTitle) {
+      return I18n.t("sidebar.sections.custom.title.validation.maximum");
+    }
   }
 
   get titleCssClass() {
     return this.title === undefined || this.validTitle ? "" : "warning";
   }
+
+  get #blankTitle() {
+    return isEmpty(this.title);
+  }
+
+  get #tooLongTitle() {
+    return this.title.length > 30;
```

Comment:
I think we need to declare `30` as a constant on the server side and then use the `javascript:update_constants` rake task to expose this constant to the client side. Otherwise, we're currently duplicating this number across multiple places like the client side, server side, system tests and translations making it hard to change this number in the future.